overlord/ifacestate/ifacestate_test.go

```diff
@@ -233,6 +233,39 @@ func (s *interfaceManagerSuite) TestDisconnectConflictsSlotSnap(c *C) {
 	s.testConnectDisconnectConflicts(c, ifacestate.Disconnect, "producer")
 }
 
+func (s *interfaceManagerSuite) TestConnectDoesntConflict(c *C) {
```

Comment:
I think we want a simple test also that if there's a connect/disconnect change,  snapstate.Update returns "snap has changes". See snapstate_test.go for examples of such tests